pkg/controller/sync/aggregator/aggregator.go

```diff
@@ -0,0 +1,105 @@
+package aggregator
+
+import (
+	"fmt"
+	gosync "sync"
+
+	"golang.org/x/exp/maps"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type KindName struct {
+	Kind string
+	Name string
+}
+
+func NewGVKAggregator(allowedKeyKinds []string) *GVKAgreggator {
```

Comment:
do we envision a world where the allowedKeyKinds change across instantiations of the GVKAggregator?  What is the need for this polymorphism?

Reply:
IMO this is unnecessary: the caller bears the responsibility of providing a valid range of input values.

Otherwise, not only does this complicate the interface (errors are possible where otherwise they wouldn't be), but also extends the role of aggregator beyond a simple maintainer of sets.

Reply:
Max and I had an interesting conversation about what level of responsibility the SyncAggregator should bare in terms of regulating its inputs.  

Based on that conversation, it feels to me that having "allowed kinds" and validating them seems to mix together two ideas:

1. package consumer can be trusted to provide valid keys
2. package consumer cannot be trusted to provide valid keys

IMO it could be helpful to go all-in on either 1 or 2.

1 would be: make the key simply a string.  The consumer is responsible for writing its keys correctly and the SyncAggregator cares not what kind of keys are passed (as long as they are strings).  The consumer would be wise to write its keys in a predictable way that doesn't overlap with other consumers.  This makes sense when the consumer is skillful and a known entity (I.e. us)

2 would be: The SyncAggregator is hardcoded to expect the actual `Config` resource struct or the `SyncSet` resource struct.  This is less flexible but type safe and rigid.  Personally this one feels like overkill.  This could be helpful if the SyncAggregator might be open to the world and we want to prevent having to support things that were outside of our intended scope.  That doesn't feel like a problem we actually have.

For now I'd probably just remove the validation and move forward.  We can always add more rigidity later if we're worried things will be misused.

Reply:
I've taken out the key validation piece; we can add it later if we need it.